carbon-design-system/carbon
# fix(tag): pass props.disabled to close button

Closes #5995 

This passes `props.disabled` to the `button` element in filter tags to prevent firing `onClose`. Also fixes `pointer` cursor on hover.

Currently, the filter tag still emits `onClick` because the body element is a `div` instead of a `button`. This was introduced in #5470 since the tag body shouldn't be a click target anymore. If I understand correctly this means `onClick` will be deprecated, right?

#### Changelog

**New**

- `disabled` attribute on `button.bx--tag__close-icon`
- explicit `cursor: not-allowed` on `.bx--tag__close-icon:disabled`

#### Testing / Reviewing

- Verify `onClose` is not firing when the close button is clicked on a disabled tag
- Verify cursor on close button hover when disabled
